flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/state/rocksdb/restore/DistributeStateHandlerHelper.java

```diff
@@ -0,0 +1,176 @@
+
+        // Open database using restored instance helper method
+        this.restoredDbInstance =
+                RocksDBRestoreOperationUtils.restoreTempDBInstanceFromLocalState(
+                        stateHandle,
+                        stateMetaInfoSnapshots,
+                        columnFamilyOptionsFactory,
+                        dbOptions,
+                        ttlCompactFiltersManager,
+                        writeBufferManagerCapacity);
+    }
+
+    /**
+     * Distributes state handle data by checking SST file ranges and exporting column families.
+     * Returns the key group range if successful, null if the handle was skipped.
+     *
+     * @param exportCfBasePath base path for export
+     * @param exportedColumnFamiliesOut output parameter for exported column families
+     * @param skipped output parameter for skipped handles
+     * @return key group range if successfully exported, null otherwise
+     * @throws Exception on any export error
+     */
+    public KeyGroupRange distribute(
+            Path exportCfBasePath,
+            Map<RegisteredStateMetaInfoBase.Key, List<ExportImportFilesMetaData>>
+                    exportedColumnFamiliesOut,
+            List<IncrementalLocalKeyedStateHandle> skipped)
+            throws Exception {
+
+        final String logLineSuffix = createLogLineSuffix();
+
+        List<ColumnFamilyHandle> tmpColumnFamilyHandles = restoredDbInstance.columnFamilyHandles;
+
+        LOG.debug("Checking actual keys of sst files" + logLineSuffix);
+
+        // Check SST file range
+        RocksDBIncrementalCheckpointUtils.RangeCheckResult rangeCheckResult =
+                RocksDBIncrementalCheckpointUtils.checkSstDataAgainstKeyGroupRange(
+                        restoredDbInstance.db, keyGroupPrefixBytes, stateHandle.getKeyGroupRange());
+
+        LOG.info("{}" + logLineSuffix, rangeCheckResult);
+
+        if (rangeCheckResult.allInRange()) {
+            LOG.debug("Start exporting" + logLineSuffix);
+
+            List<RegisteredStateMetaInfoBase> registeredStateMetaInfoBases =
+                    restoredDbInstance.stateMetaInfoSnapshots.stream()
+                            .map(RegisteredStateMetaInfoBase::fromMetaInfoSnapshot)
+                            .collect(Collectors.toList());
+
+            // Export all the Column Families and store the result in exportedColumnFamiliesOut
+            RocksDBIncrementalCheckpointUtils.exportColumnFamilies(
+                    restoredDbInstance.db,
+                    tmpColumnFamilyHandles,
+                    registeredStateMetaInfoBases,
+                    exportCfBasePath,
+                    exportedColumnFamiliesOut);
+
+            LOG.debug("Done exporting" + logLineSuffix);
+            return stateHandle.getKeyGroupRange();
+        } else {
+            skipped.add(stateHandle);
+            LOG.debug("Skipped export" + logLineSuffix);
+            return null;
```

Comment:
It'd more clear to me if  we collect skipped handles on the caller side; 
and return from this method  `Either<KeyGroupRange, IncrementalLocalKeyedStateHandle>`.

WDYT?